### PR TITLE
[3.7] Make maximum query parallelism configurable.

### PR DESCRIPTION
**Pull request by danielhlarkin**
Backport of https://github.com/arangodb/arangodb/pull/12234

### CODE REVIEW

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11110/

File: arangod/Aql/SharedQueryState.h [hunk 1]
```diff
@@ -141,7 +146,7 @@ class SharedQueryState final : public std::enable_shared_from_this<SharedQuerySt
   unsigned _cbVersion; // increased once callstack is done
   
   // TODO: make configurable
```
Comment (jsteemann) [inline arangod/Aql/SharedQueryState.h:148]: the comment is now obsolete
```suggestion
```

**Comment by danielhlarkin** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/11112/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**